### PR TITLE
Multiple cfgs

### PR DESCRIPTION
**Pull request by awirb**
If using different modules in a new function
- in the ini file for this function, add a section called modName_override for each module
- configurations for these modules will be created using the (local if defaultConfig=False) default configuration and overriding the parameters from the modName_override section
- in the end, a function_settings.ini is saved to the outputs folder of the function:  all the other parameters, which were not part of the override parameters, are added in the override sections for easy reproduction

### CODE REVIEW

File: avaframe/com3Hybrid/com3Hybrid.py [hunk 1]
```diff
@@ -41,8 +43,15 @@ def maincom3Hybrid(cfgMain, cfgHybrid):
...
-    hybridModelDFACfg = pathlib.Path('com3Hybrid', 'hybridModel_com1DFACfg.ini')
+
+    # setup outputs folder
+    oPath = pathlib.Path(avalancheDir, 'Outputs', 'com3Hybrid')
+    fU.makeADir(oPath)
+
+    # get comDFA configuration and save to file
+    hybridModelDFACfg = cfgUtils.getModuleConfig(com1DFA, fileOverride='', modInfo=False, toPrint=False,
+        onlyDefault=cfgHybrid['com1DFA_override']['defaultConfig'], overrideConfig=cfgHybrid)
+    hybridModelDFACfgFile = cfgUtils.writeCfgFile(avalancheDir, com1DFA, hybridModelDFACfg, fileName='com1DFA_settings', filePath=oPath)
```
Comment (matthiasto) [inline avaframe/com3Hybrid/com3Hybrid.py:54]: why write this one and not also the AB and pathGeneration cfgs to file?

**Review by matthiasto — CHANGES_REQUESTED** [review]

File: avaframe/com3Hybrid/com3Hybrid.py [hunk 2]
```diff
@@ -41,8 +43,15 @@ def maincom3Hybrid(cfgMain, cfgHybrid):
...
     demOri = getInput.readDEM(avalancheDir)
-    # get comDFA configuration path for hybrid model
-    hybridModelDFACfg = pathlib.Path('com3Hybrid', 'hybridModel_com1DFACfg.ini')
+
+    # setup outputs folder
+    oPath = pathlib.Path(avalancheDir, 'Outputs', 'com3Hybrid')
+    fU.makeADir(oPath)
+
+    # get comDFA configuration and save to file
+    hybridModelDFACfg = cfgUtils.getModuleConfig(com1DFA, fileOverride='', modInfo=False, toPrint=False,
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com3Hybrid/com3Hybrid.py:52]: what is your suggestion? I think it is actually pretty accurate the one thing is a file path that you have to provide the other is configparser object that has the override parameters in it..

File: avaframe/com3Hybrid/com3Hybrid.py [hunk 3]
```diff
@@ -54,21 +63,25 @@ def maincom3Hybrid(cfgMain, cfgHybrid):
...
 
         # Run dense flow with coulomb friction
-        dem, _, _, simDF = com1DFA.com1DFAMain(avalancheDir, cfgMain, cfgFile=hybridModelDFACfg)
+        dem, _, _, simDF = com1DFA.com1DFAMain(avalancheDir, cfgMain, cfgFile=hybridModelDFACfgFile)
         simID = simDF.index[0]
         particlesList, timeStepInfo = particleTools.readPartFromPickle(avalancheDir, simName=simID, flagAvaDir=True,
                                                                        comModule='com1DFA')
+
+        # fetch configuration for DFAPathGeneration
+        hybridModelPathCfg= cfgUtils.getModuleConfig(DFAPath, fileOverride='', modInfo=False, toPrint=False,
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/com3Hybrid/com3Hybrid.py:79]: this is a good point!

**Comment by awirb** [comment]
newly opened in PR #740

**Pull request closed without merging**